jib-cli/src/test/java/com/google/cloud/tools/jib/cli/cli2/logging/CliLoggerTest.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.cli.cli2.logging;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.cloud.tools.jib.api.LogEvent;
+import com.google.cloud.tools.jib.plugins.common.logging.ConsoleLogger;
+import com.google.cloud.tools.jib.plugins.common.logging.SingleThreadedExecutor;
+import java.io.PrintStream;
+import java.time.Duration;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.EnvironmentVariables;
+import org.junit.contrib.java.lang.system.RestoreSystemProperties;
+import org.junit.runner.RunWith;
+import org.mockito.InOrder;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/** Tests for {@link CliLogger}. */
+@RunWith(MockitoJUnitRunner.class)
+public class CliLoggerTest {
+
+  @Rule public RestoreSystemProperties restoreSystemProperties = new RestoreSystemProperties();
+  @Rule public EnvironmentVariables environmentVariables = new EnvironmentVariables();
+
+  @Mock PrintStream mockOut;
+  @Mock PrintStream mockErr;
+
+  @Mock CliLogger mockCliLogger;
+
+  private void sendMessages(CliLogger logger) {
+    logger.debug("debug");
+    logger.info("info");
+    logger.lifecycle("lifecycle");
+    logger.warn("warn");
+    logger.error("error");
+  }
+
+  @Test
+  public void testLog_quiet() {
+    CliLogger logger = new CliLogger(Verbosity.quiet, mockOut, mockErr);
+    sendMessages(logger);
+    Mockito.verifyNoInteractions(mockOut);
+    Mockito.verifyNoInteractions(mockErr);
+  }
+
+  @Test
+  public void testLog_error() {
+    CliLogger logger = new CliLogger(Verbosity.error, mockOut, mockErr);
+    sendMessages(logger);
+    Mockito.verify(mockErr).println("error");
+    Mockito.verifyNoMoreInteractions(mockErr);
+    Mockito.verifyNoInteractions(mockOut);
+  }
+
+  @Test
+  public void testLog_warn() {
+    CliLogger logger = new CliLogger(Verbosity.warn, mockOut, mockErr);
+    sendMessages(logger);
+    Mockito.verify(mockErr).println("error");
+    Mockito.verifyNoMoreInteractions(mockErr);
```

Comment:
`verifyNoMoreInteractions` tend to be used sparingly. Can I have some context into why we need to call it here?

Reply:
In this case, you want to verify that (1) There's no mixup between mockOut and mockErr and (2) There's no extra logging happening to the printstreams.